### DISCUSSION TITLE
Distribution of agents in areas with obstacles

### DISCUSSION THREAD

**Discussion by chraibi** [PedestrianDynamics/jupedsim#1308] [General]
When setting up agent distribution in specific areas, it's crucial to ensure that spawning zones don't overlap with obstacles. 

Consider a scenario where agents are evenly spread across an entire geometry, including obstacles. This is achieved effectively using the following function, which treats obstacles as part of the geometry:

`
distribute_agents(spawning_area=walkable_area)
`

![distribution2](https://github.com/PedestrianDynamics/jupedsim/assets/5772973/0005cc2c-6482-47d5-a13a-b634be6895da)

In contrast, a more common situation involves distributing agents within a designated spawning_area. An example of this would be:

`
distribute_agents(spawning_area=spawning_area)
`

﻿﻿This method, however, may result in agents appearing inside obstacles.

![distribution1](https://github.com/PedestrianDynamics/jupedsim/assets/5772973/2adc4b13-6ac2-414d-b4de-6b3fd14479a4)

The question arises: Should the jps.distribution functions automatically account for obstacles within spawning areas? Or is it more appropriate for users to meticulously select spawning areas, even though it may be less convenient?

**Comment by Ozaq** [comment]
This can be achieved by taking the intersection of your rectangle and the walkable area, see: https://shapely.readthedocs.io/en/stable/reference/shapely.intersection.html#shapely.intersection

Example:
```
distribute_agents(spawning_area=shapely.intersection(spawning_area, walkable_area))
```
